src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.service.search;
+
+import bio.terra.common.exception.PdaoException;
+import bio.terra.model.SearchIndexModel;
+import bio.terra.model.SearchIndexRequest;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.CreateIndexResponse;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.elasticsearch.client.indices.PutMappingRequest;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class SearchService {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final RestHighLevelClient client;
+
+    @Autowired
+    public SearchService(BigQueryPdao bigQueryPdao) {
+        // needs to be moved into own class with config
+        final RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200));
+        this.bigQueryPdao = bigQueryPdao;
+        this.client = new RestHighLevelClient(builder);
+    }
+
+    private String createEmptyIndex(Snapshot snapshot) {
+        String indexName = String.format("idx-%s", snapshot.getId());
+        try {
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest(indexName)
+                .settings(Settings.builder()
+                    .put("index.number_of_shards", 3));
+            CreateIndexResponse response = client.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+            if (!response.isAcknowledged()) {
+                throw new PdaoException("The index request was not acknowledged by one or more nodes");
```

Comment:
I'm not familiar with this part of ES. If a request isn't acknowledged by all nodes, is that a fatal error? Although, this can wait until we have a better idea about how things fail typically, and how to handle those failures.

Reply:
It isn't possible to identify which nodes failed from this method: https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-create-index.html#java-rest-high-create-index-response .

Reply:
The thing about both of those fields is that they only tell us if the server timed out waiting for the ack. They don't tell us that the operation didn't eventually complete. But maybe in practice this error is very rare, if the ES nodes aren't under heavy load.